src/main/java/nl/dtls/fairdatapoint/service/impl/FairSearchClientImpl.java

```diff
@@ -62,17 +64,18 @@
      * @param uri   FDP base URI
      * @return Http response code
      */
+    
+    @Async
     @Override
-    public int submitFdpUri(@Nonnull IRI uri) {
+    public CompletableFuture submitFdpUri(@Nonnull IRI uri) {
         Preconditions.checkState(uri != null, "FDP uri can't be null");
         try {
-            HttpResponse<String> response = Unirest.get(fdpSubmitUrl).header("fdp", uri.toString())
-                    .asString();
-            return response.getStatus();
+            return CompletableFuture.completedFuture(Unirest.get(fdpSubmitUrl).queryString("fdp", 
```

Comment:
`CompletableFuture.completedFuture` is run synchronous, **not asynchronous**. For async behaviour (which I believe is what you try to achieve here), you can use a call like [`runAsync`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#runAsync-java.lang.Runnable-java.util.concurrent.Executor-).